### PR TITLE
Replace circleci/mysql image with the official mysql docker image

### LINKED ISSUE #18951: Update CircleCI docker images
Follow-up from https://github.com/mozilla/addons-server/issues/18843 we're also using on addons-server some legacy convenience images:
https://github.com/mozilla/addons-server/blob/fde1bb0f1dfce82a6a9459e68ab9f566f44bf8e1/.circleci/config.yml#L11
https://github.com/mozilla/addons-server/blob/fde1bb0f1dfce82a6a9459e68ab9f566f44bf8e1/.circleci/config.yml#L288

Those are [deprecated](https://discuss.circleci.com/t/legacy-convenience-image-deprecation/41034) and we need to update them to supported images. The modern CircleCI convenience images have the `cimg/` prefix (but we need to check if the ones we need exist or need to be replaced with some upstream image instead)

### PR DESCRIPTION
**Pull request by diox**
Downgrading (!) the version too to match prod.

Fixes #18951

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -6,9 +6,10 @@
 version: 2.1
 
 orbs:
-  # This is needed to be able to install Node.js in the primary container,
-  # which provides python only.
-  node: circleci/node@4.7.0
+  # This is needed to be able to install Node.js 14 in the primary container,
+  # which provides python only (the convenience cimg/python:3.9-node provides
+  # node 16).
+  node: circleci/node@5.0.0
```
Comment (diox) [inline .circleci/config.yml:12]: This is just an orb, so it doesn't need to change. I updated the version but it shouldn't be any different, we're explicitly installing a specific nodejs version.

Comment (eviljeff) [reply .circleci/config.yml:12]: _could_ it change though?  Using the deprecated images seems undesirable, even if it's not an immediate problem.

Comment (diox) [reply .circleci/config.yml:12]: It's (confusingly) not an image, just a (very fancy) bit of config. It isn't deprecated like the `circleci/` images are, and doesn't have a replacement.

Comment (eviljeff) [reply .circleci/config.yml:12]: Oh! got it

**Review by bobsilverberg — APPROVED** [review]
Seems fine to me. I assume it works or CI would have failed.

**Pull request merged**